### DISCUSSION TITLE
Width of  text edit boxes

### DISCUSSION THREAD

**Discussion by John-Nagle** [emilk/egui#2173] [Q&A]
![logindialog](https://user-images.githubusercontent.com/942608/196855353-58d4b90d-5324-4392-81f9-aa9396b978c8.png)

My text edit boxes are always the minimum width. Draw function looks like this;

    fn draw(&mut self, ctx: &egui::Context) {
        if self.is_open {
            let mut dismissed = false;          // true if dismiss button pushed
            let window = egui::containers::Window::new(self.title.as_str()).id(self.id)
                .collapsible(false)
                .anchor(egui::Align2::CENTER_CENTER, [0.0, 0.0]);
            window.show(ctx, |ui| {             
                egui::Grid::new("login box").show(ui, |ui| {   
                    ui.label("Username");
                    let _ = ui.add(egui::TextEdit::singleline(&mut self.login_auth_params.user_name).desired_width(200.0));
                    ui.end_row();
                    ui.label("Password");
                    let _ = ui.add(egui::TextEdit::singleline(&mut self.login_auth_params.password).password(true));
                    ui.end_row();
   	            });
                
                ui.vertical_centered(|ui| {
                    //  ***MORE*** need access to state***
                    ////if ui.add(egui::Button::new(t!("menu.cancel", state.get_lang()))).clicked() {
                    let filled_in = self.login_auth_params.is_filled_in();  // if form filled in
                    if ui.add_enabled(filled_in, egui::Button::new("Login")).clicked() {
                    ////if ui.add(egui::Button::new("Login")).clicked() {
                        dismissed = true;                       // dismiss
                        //  ***NEED TO DO SOMETHING TO START LOGIN PROCESS***
                    }
                });
            });
            if dismissed { self.is_open = false; } // do here to avoid borrow clash
        }
    }

Note the line 

    let _ = ui.add(egui::TextEdit::singleline(&mut self.login_auth_params.user_name).desired_width(200.0));

which ought to force a wider edit box, but doesn't. What am I doing wrong here?

Egui 0.18.1, not 0.19. because Rend3 hasn't advanced to 0.19 yet.

**Comment by John-Nagle** [comment]
Turns out that **egui::Grid** does not widen cells just because **TextEdit** wants more space.  Unsure if this is a bug or intended behavior.

**Comment by pietervandermeer** [comment]
Use ui.add_sized() This is also mentioned in the API documentation.
